Tools/clinic/clinic.py

```diff
@@ -4474,7 +4466,11 @@ def pre_render(self) -> None:
     @property
     def parser_type(self) -> str:
         assert self.type is not None
-        return required_type_for_self_for_parser(self.function) or self.type
+        f = self.function
+        if f.kind in (METHOD_INIT, METHOD_NEW, STATIC_METHOD, CLASS_METHOD):
+            tp, _ = correct_name_for_self(f)
+            return tp
+        return self.type
```

Comment:
Nitpicks:
- Not sure we really need the intermediate variable
- I'd use a set rather than a tuple

```suggestion
        if self.function.kind in {METHOD_INIT, METHOD_NEW, STATIC_METHOD, CLASS_METHOD}:
            tp, _ = correct_name_for_self(self.function)
            return tp
        return self.type
```

Reply:
Thanks!